src/streaming/controllers/BufferController.js

```diff
@@ -553,7 +564,12 @@ function BufferController(config) {
     }
 
     function notifyBufferStateChanged(state) {
-        if (bufferState === state || (type === Constants.FRAGMENTED_TEXT && !textController.isTextEnabled())) return;
+        if (bufferState === state ||
+            (state === BUFFER_EMPTY && playbackController.getTime() === 0) || // Don't trigger BUFFER_EMPTY if it's initial loading
```

Comment:
it's only for VOD streams?

Reply:
I think it should work in every stream, that check is only to avoid triggering BUFFER_EMPTY while loading the first segments